eval/exception.go

```diff
@@ -38,12 +38,28 @@ func (exc *Exception) Pprint(indent string) string {
 	} else {
 		msg = "\033[31;1m" + exc.Cause.Error() + "\033[m"
 	}
-	fmt.Fprintf(buf, "Exception: %s\n", msg)
-	buf.WriteString(indent + "Traceback:")
+	fmt.Fprintf(buf, "Exception: %s", msg)
+
+	// In case of a simple error generated from only one line
+	// don't print the full traceback format by setting
+	// printContext to false.
+	printContext := false
+	if tb := exc.Traceback; tb != nil {
+		if tb.Name != "[interactive]" {
```

Comment:
Having "[interactive]" here doesn't look good, could it be turn into a constant?

Reply:
Add an additional field to `util.SourceContext` to indicate whether the source is an interactive session (say `tb.Interactive`).

I would prefer you invert the logics here - instead of keeping a `printContext` flag, keep a `abbrTraceback` flag:

```go
// Abbreviate the traceback when it only contains one entry and that entry is from the interactive session
abbrTraceback := exc.Traceback.Interactive && exc.Traceback.Next == nil
```